src/lib/useDebouncedInputValue.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useRef, useState } from "react";
+import { useLatest } from "react-use";
+
+const DEBOUNCE_MS = 50;
+
+export function useDebouncedInputValue<T>(defaultState: T, commit: (value: T) => void) {
+  const [value, setValue] = useState(defaultState);
+  const timeoutRef = useRef<number>();
+  const latestValue = useLatest(value);
+
+  useEffect(() => {
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
+    }
+
+    timeoutRef.current = window.setTimeout(() => {
+      commit(value);
+    }, DEBOUNCE_MS);
+  });
```

Comment:
Correct me if I'm wrong, but I think there may be a potential problem here due to the lack of deps array. In the event of permanent rerendering, we could get stuck in a situation where the commit never triggers.

Reply:
wow, that's a catch